packages/client/src/client.js

```diff
@@ -67,6 +68,11 @@ export default class PercyClient {
 
   // Stringifies client and environment info.
   userAgent() {
+    if (!this.envWarning && (this.clientInfo.size === 0 || this.environmentInfo.size === 0)) {
+      this.envWarning = true;
+      this.log.warn('Warning: Missing `clientInfo` and/or `environmentInfo` properties');
+    }
+
     let client = [`Percy/${/\w+$/.exec(this.apiUrl)}`]
       .concat(`${pkg.name}/${pkg.version}`, ...this.clientInfo)
       .filter(Boolean).join(' ');
```

Comment:
The forced client info here is safe to assume will always be unique since no other SDK should be providing API or `@percy/client` info. However, if we wanted to be sure it is unique, we'd have to do the same as my suggestion below, only because of the order of forced client info, we would also need to reverse the resulting array.

(Again, I can't comment on unchanged folded code below)

For environment info below, we can add the forced info to the Set so it's always unique. I would suggest cloning the set to avoid mutation, but considering the whole point of Sets is to have unique values, it's safe to just keep adding the forced info blindly. We still want the variable to be an array though, so we can just move the `concat` into `[...envInfo.add(...).add(...)]`. (I really wish GH would let me comment on folded code)

Reply:
Also, if we forcibly add both client and env info, the `envWarning` won't be needed since we know an empty value would only ever happen the first time